File: CSharp Basics/Homeworks/Console-Input-Output-Homework/06.QuadraticEquation/QuadraticEquation.cs
using System;
// float and double's are entered with "," in the console
class QuadraticEquation
{
    static void Main(string[] args)
    {
        Console.WriteLine("Please enter a:");
        double a = double.Parse(Console.ReadLine());
        Console.WriteLine("Please enter b:");
        double b = double.Parse(Console.ReadLine());
        Console.WriteLine("Please enter c:");
        double c = double.Parse(Console.ReadLine());
        double firstRoot = (-b - Math.Sqrt(Math.Pow(b, 2) - 4 * a * c)) / (2 * a);
        double secondRoot = (-b + Math.Sqrt(Math.Pow(b, 2) - 4 * a * c)) / (2 * a);
        if (firstRoot.ToString() == "NaN" && secondRoot.ToString() == "NaN")
        {
            Console.WriteLine("no real roots");
        }
        else
        {
            Console.WriteLine("x1 = {0}, x2 = {1}", firstRoot, secondRoot);
        }
    }
}

File: JavaScript Basics/Homeworks/02.JavaScript-Syntax/01.QuadraticEquation/QuadraticEquation.js
//double firstRoot = (-b - Math.Sqrt(Math.Pow(b, 2) - 4 * a * c)) / (2 * a);
//double secondRoot = (-b + Math.Sqrt(Math.Pow(b, 2) - 4 * a * c)) / (2 * a);
//if (firstRoot.ToString() == "NaN" && secondRoot.ToString() == "NaN") {
//    Console.WriteLine("no real roots");
//}
//else {
//    Console.WriteLine("x1 = {0}, x2 = {1}", firstRoot, secondRoot);
//}
function getRoots(a, b, c) {
    a = parseInt(a);
    b = parseInt(b);
    c = parseInt(c);
    var firstRoot = (-b - Math.sqrt(Math.pow(b, 2) - 4 * a * c)) / (2 * a);
    var secondRoot = (-b + Math.sqrt(Math.pow(b, 2) - 4 * a * c)) / (2 * a);
    if (!firstRoot || !secondRoot) {
        console.log("no real roots");
    }
    else {
        if (firstRoot === secondRoot) {
            console.log("x = " + firstRoot);
        }
        else {
            console.log("x1 = " + firstRoot + " x2 = " + secondRoot);
        }
    }
}

getRoots(2, 5, -3);
getRoots(2, -4, 2);
getRoots(4, 2, 1);